Printing

bmmoore opened the pull request:
Fixes  #167

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

virgil-serbanuta commented on src/main/haskell/kore/src/Text/IndentingPrinter.hs:
```diff
@@ -1,43 +1,42 @@
+      doc = execWriter printer
+      simpleStream = layoutPretty (defaultLayoutOptions {layoutPageWidth = Unbounded}) doc
+
+docToString :: Doc ann -> String
```
I know that the other public functions here do not have documentation, but they were written before the current style existed. Could you document this one?

bmmoore commented:
@virgil-serbanuta I added comments to the new function, and also noticed that the `IsString ShowS` instance was not used anymore during the first commit.

Can you add some comments on other functions?

virgil-serbanuta commented:
Sure, I added them.

bmmoore commented:
Ah, the build was failing over some warnings I missed by not building with `--pedantic`. I dropped an unused import in IndentingPrinter, and renamed a few shadowing arguments from `list` to `items`.
